zebra-network/src/config.rs

```diff
@@ -8,38 +12,81 @@ use crate::network::Network;
 pub struct Config {
     /// The address on which this node should listen for connections.
     pub listen_addr: SocketAddr,
+
     /// The network to connect to.
     pub network: Network,
+
     /// The user-agent to advertise.
     pub user_agent: String,
-    /// A list of initial peers for the peerset.
-    ///
-    /// XXX this should be replaced with DNS names, not SocketAddrs
-    pub initial_peers: Vec<SocketAddr>,
+
+    /// A list of initial peers for the peerset when operating on
+    /// mainnet.
+    pub initial_mainnet_peers: Vec<String>,
+
+    /// A list of initial peers for the peerset when operating on
+    /// testnet.
+    pub initial_testnet_peers: Vec<String>,
+
     /// The outgoing request buffer size for the peer set.
     pub peerset_request_buffer_size: usize,
 
     // Note: due to the way this is rendered by the toml
     // serializer, the Duration fields should come last.
     /// The default RTT estimate for peer responses, used in load-balancing.
     pub ewma_default_rtt: Duration,
+
     /// The decay time for the exponentially-weighted moving average response time.
     pub ewma_decay_time: Duration,
+
     /// The timeout for peer handshakes.
     pub handshake_timeout: Duration,
+
     /// How frequently we attempt to connect to a new peer.
     pub new_peer_interval: Duration,
 }
 
+impl Config {
+    fn parse_peers<S: ToSocketAddrs>(peers: Vec<S>) -> Vec<SocketAddr> {
+        peers
+            .iter()
+            .flat_map(|s| s.to_socket_addrs())
+            .flatten()
+            .collect::<Vec<SocketAddr>>()
+    }
+
+    /// Get the initial seed peers based on the configured network.
+    pub fn initial_peers(&self) -> Vec<SocketAddr> {
+        match self.network {
+            Network::Mainnet => Config::parse_peers(self.initial_mainnet_peers.clone()),
+            Network::Testnet => Config::parse_peers(self.initial_testnet_peers.clone()),
+        }
+    }
```

Comment:
take-or-leave-it: since `parse_peers` is never used outside of `initial_peers`, it could be folded in as `match self.network { ... } .iter().flat_map(...).flatten().collect()`, and this saves an allocation by only doing one `collect` instead of two.